### PR TITLE
LUCENE-9705: Create Lucene90PointsFormat

### PR DESCRIPTION
**Pull request by iverase**
For now this is just a copy of Lucene86PointsFormat. The existing Lucene86PointsFormat is moved to backwards-codecs.

This is actually the last codec that needs to be moved to Lucene90 :)

### CODE REVIEW

**Review by jtibshirani — COMMENTED** [review]
As you mentioned, this is the only codec we're moving that still shares a chunk of logic with older codecs (through `BKDWriter` and `BKDReader`). A couple questions:
* How does this affect your planned work in [LUCENE-9047](https://issues.apache.org/jira/browse/), which was a big motivation for creating these new codecs? Will you just add another 'internal version' for the endianness change?
* Confirming my understanding, we plan to follow up with something like [LUCENE-9820](https://issues.apache.org/jira/browse/LUCENE-9820) to avoid having to share so many classes with older codecs in the future?

File: lucene/core/src/test/org/apache/lucene/codecs/lucene90/TestLucene90PointsFormat.java [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+  }
+
+  @Override
+  public void testMergeStability() throws Exception {
+    assumeFalse(
+        "TODO: mess with the parameters and test gets angry!", codec instanceof FilterCodec);
+    super.testMergeStability();
+  }
+
+  public void testEstimatePointCount() throws IOException {
```
[the remark this answers is not in the document]
Comment (iverase) [reply lucene/core/src/test/org/apache/lucene/codecs/lucene90/TestLucene90PointsFormat.java:101]: I moved some test but the ones that relay on knowing the maxPointsInLeafNode cannot be moved as the information is not expose through the PointValues API.

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_codecs/lucene86/Lucene86PointsWriter.java [hunk 2]
```diff
@@ -60,40 +60,40 @@ public Lucene86PointsWriter(
         IndexFileNames.segmentFileName(
             writeState.segmentInfo.name,
             writeState.segmentSuffix,
-            Lucene86PointsFormat.DATA_EXTENSION);
+            Lucene86RWPointsFormat.DATA_EXTENSION);
```
[the remark this answers is not in the document]
Comment (iverase) [reply lucene/backward-codecs/src/test/org/apache/lucene/backward_codecs/lucene86/Lucene86PointsWriter.java]: good catch

**Comment by iverase** [comment]
> How does this affect your planned work in LUCENE-9047, which was a big motivation for creating these new codecs? Will you just add another 'internal version' for the endianness change?

I don't think we will need to add a new internal version. I think for backwards codecs we can wrap the IndexInputs so it would be transparent for the BKD reader.

> Confirming my understanding, we plan to follow up with something like LUCENE-9820 to avoid having to share so many classes with older codecs in the future?

I hope so :). What I am trying to avoid is to have to rewrite many BKD readers if we decide to change the Points API (for example [LUCENE-9619](https://issues.apache.org/jira/browse/LUCENE-9619)). So separating reading a specific Index format from the API should help.

**Review by jtibshirani — APPROVED** [review]

**Pull request merged**